### DISCUSSION TITLE
issue when using slint

### DISCUSSION THREAD

**Discussion by Dangerise** [helix-editor/helix#10740] [Troubleshooting]
`slint` is a native gui crate ,which uses its own `slint` programming language to write ui components, `slint` language can be compiled to rust code and be included by macro provided by slint crate

Now I have a slint file, which define a component, then I rename the component, then I expect the new name of the component can be found in the auto completion of the editor

In vsc, it works well

but in helix, it doesn't work I expected

https://github.com/helix-editor/helix/assets/88434542/41b0c14d-b7cc-43ca-a649-65cd615ee90a

https://github.com/helix-editor/helix/assets/88434542/ce797644-36ea-467f-9e8e-ca9d7341c124

**Comment by TornaxO7** [comment]
I can't watch the video files which you provided.
Have you tried to save your changes after renaming the component? Sometimes the LSP gets updated after executing a `save`.

**Comment by Dangerise** [comment]
@TornaxO7 yes

I update the video again

**Comment by Dangerise** [comment]
hoping to find answers

**Comment by kirawi** [comment]
Run `hx -v` and try again. Then, report what you see in `:log-open`

**Comment by Dangerise** [comment]
@kirawi 
[log.txt](https://github.com/helix-editor/helix/files/15369249/log.txt)

**Reply by kirawi** [reply]
I'm not really sure how Slint does it, but I suspect that rust-analyzer isn't updating its index for the auto-generated files because Helix doesn't implement a file watcher yet: https://github.com/helix-editor/helix/issues/1125
